Assets/PC2D/Scripts/PlatformerMotor2D.cs

```diff
@@ -3321,5 +3275,155 @@ void OnDrawGizmosSelected()
         }
     }
 
+    private void ChangeState(MotorState newState) {
+        // no change...
+        if (motorState == newState)
+        {
+            return;
+        }
+
+        if (IsSlipping())
+        {
+            if (onSlippingEnd != null)
+            {
+                onSlippingEnd();
+            }
+        }
+        if (IsJumping())
+        {
+            if (onJumpEnd != null)
+            {
+                onJumpEnd();
+            }
+        }
+        if (IsDashing())
+        {
+            if (onDashEnd != null)
+            {
+                onDashEnd();
+            }
+        }
+
+        // set
+        motorState = newState;
+
+        if (IsSlipping())
+        {
+            if (onSlipping != null)
+            {
+                onSlipping();
+            }
+        }
+        if (IsJumping())
+        {
+            if (onJump != null)
+            {
+                onJump();
+            }
+        }
+        if (IsDashing())
+        {
+            if (onDash != null)
+            {
+                onDash();
+            }
+        }
+    }
+
+    //
+    // Quering motor
+    //
+    #region Public
+    public bool IsDashing() {
+        return motorState == MotorState.Dashing;
+    }
+
+    public bool IsJumping() {
+        return motorState == MotorState.Jumping;
+    }
+
+    public bool IsUserHandled() {
+        return motorState == MotorState.FreedomState;
+    }
+
+    public bool IsFalling() {
+        return motorState == MotorState.Falling;
+    }
+
+    public bool IsFallingFast() {
+        return motorState == MotorState.FallingFast;
+    }
+
+    public bool IsWallSticking()
+    {
+        return motorState == MotorState.WallSticking;
+    }
+
+    public bool IsOnGround()
+    {
+        return motorState == MotorState.OnGround;
+    }
+
+    public bool IsSlipping() {
+        return motorState == MotorState.Slipping;
+    }
+
+    public bool IsOnCorner()
+    {
+        return motorState == MotorState.OnCorner;
+    }
+
+    public bool IsWallSliding()
+    {
+        return motorState == MotorState.WallSliding;
+    }
+
+    public bool IsInAir()
+    {
+        return IsJumping() || IsFalling() || IsFallingFast();
+    }
+
+    public bool IsOnWall()
+    {
+        return IsWallSliding() || IsOnCorner() || IsWallSticking();
+    }
+
+    public bool IsGrounded()
+    {
+        return (HasFlag(CollidedSurface.Ground) || onSlope) &&
+               !IsJumping() &&
+               (onSlope && Vector2.Dot(_velocity, slopeNormal) <= NEAR_ZERO ||
+               _velocity.y <= NEAR_ZERO);
+    }
+
+    // on slope that cannot walk, will be forced to slip down
+    // TODO improve this name
+    public bool IsForceSlipping()

```

Comment:
While reading the code, I saw IsSlipping to not be really "motor is slipping" and more like "it will slip"
So IsForceSlipping -> prerequisites to slip
IsSlipping -> when motorState is Slipping

Any idea for a better name ?